### PR TITLE
Feature/gene improvements

### PR DESCRIPTION
**Pull request by mgiacomin**
This PR adds the following improvements to GENE output reader.

- Allow the user to down sample GENE data while reading, setting `downsize > 1`. This may be useful for large GENE run.
- Fix the kx grid. In fact, GENE data are saved starting with `kx=0`, like in GS2.
- Remove the ky dependence in GENE fluxes when it reads from the `nrg` file.
- Fix an issue with the last time entry in output files. Indeed, the last time step is not always written to the output files, but it depends on whatever condition is met first between `simtimelim` and `timelim`.

### CODE REVIEW

File: pyrokinetics/gk_code/GKOutputReaderGENE.py [hunk 1]
```diff
@@ -80,7 +83,8 @@ def infer_path_from_input_file(filename: PathLike) -> Path:
         # the directory.
         filename = Path(filename)
         num_part_regex = re.compile(r"(\d{4})")
-        num_part_match = num_part_regex.search(filename.name)
+        num_part_match = num_part_regex.search(str(Path(filename).name))
```
Comment (ZedThree) [inline pyrokinetics/gk_code/GKOutputReaderGENE.py]: Were you getting an error with some input here? `filename` should already be a `Path` from L84 above, and I thought that `Path.name` was already `str`

**Review by ZedThree — APPROVED** [review]
Might be nice to have a test case that shows off how `downsize` works

File: pyrokinetics/gk_code/GKOutputReaderGENE.py (hunk 1 above)
Comment (mgiacomin) [reply pyrokinetics/gk_code/GKOutputReaderGENE.py]: I can no longer reproduce the error. Maybe when I originally forked from `unstable`, L84 was not there and appeared after a later merge with `unstable`. I will remove this change. Thank you for noticing it!

**Comment by bpatel2107** [comment]
Can we make this general to all codes? We load everything into an xarray dataset so it should be trivial to do this. The only difference is when we do this in the workflow. I'm not sure if this way saves memory or not but I think it still reads in the whole output file so it may not be much different?

We could also add a test in comparing the output from using downsample here compare to downsampling the output using xarray `.sel` method?

**Comment by mgiacomin** [comment]
The entire output file is not read when `downsize>1`. This reduces the memory usage and makes data loading faster.

This can be extended to CGYRO and GS2. I started with GENE because I have quite large output files.

**Pull request merged**